proj_eclipse/src/main/java/smartmag/models/ProductModel.java

```diff
@@ -288,6 +291,9 @@ static Prodotto prodottoFromRecord(ProdottoRecord r) {
 	 * @param r record nelquale copiare i dati del prodotto
 	 */
 	private static void copyProdottoIntoRecord(Prodotto p, ProdottoRecord r) {
+		if (r == null) {
+
+		}
```

Comment:
```suggestion
		if (r == null) {
				throw new IllegalArgumentException("Record nullo!");
		}
```
altrimenti sotto lancia _NullPointerException_